### ISSUE TITLE
google_compute_instance_from_template does not expose network info until VM exists

### ISSUE THREAD

**Issue by elfchief** [terraform-providers/terraform-provider-google#1805]
### Terraform Version
```
$ terraform -v
Terraform v0.11.7
+ provider.google v1.16.0
```

### Affected Resource(s)
- google_compute_instance_from_template

### Terraform Configuration Files
```hcl
resource "google_dns_managed_zone" "sample" {
  name        = "sample"
  dns_name    = "sample.network."
  description = "dns zone for bug reproduction purposes"
}

resource "google_compute_instance_template" "gce-minimal" {
  name        = "gce-minimal"
  description = "minimal image for low-storage, low-memory needs"

  instance_description = "description assigned to basic server instances"
  machine_type         = "n1-standard-2"
  can_ip_forward       = false

  scheduling {
    automatic_restart   = true
    on_host_maintenance = "MIGRATE"
  }

  // Create a new boot disk from an image
  disk {
    source_image = "centos-7"
    auto_delete  = true
    boot         = true
    disk_size_gb = "25"
    disk_type    = "pd-ssd"
  }

  network_interface {
    network = "default"
  }

  service_account {
    scopes = ["userinfo-email", "compute-ro", "storage-ro"]
  }
}

resource google_compute_instance_from_template "sampleXX" {
  name = "sample${format("%02d", count.index + 1)}"
  count = 2
  zone = "us-west1-a"

  source_instance_template = "${google_compute_instance_template.gce-minimal.self_link}"
}

resource "google_dns_record_set" "sampleXX" {
  name  = "sample${format("%02d", count.index + 1)}.${google_dns_managed_zone.sample.dns_name}"
  count = 2
  type  = "A"
  ttl   = 300

  managed_zone = "${google_dns_managed_zone.sample.name}"

  rrdatas = ["${element(google_compute_instance_from_template.sampleXX.*.network_interface.0.address, count.index)}"]
}
```

### Debug Output
https://gist.github.com/elfchief/0aaea3b489a1e50b89909ecfa2015a1b

### Expected Behavior
VMs should be created and then DNS entries should be created for them.

### Actual Behavior
```
Error: Error running plan: 1 error(s) occurred:

* google_dns_record_set.sampleXX: 2 error(s) occurred:

* google_dns_record_set.sampleXX[0]: Resource 'google_compute_instance_from_template.sampleXX' does not have attribute 'network_interface.0.address' for variable 'google_compute_instance_from_template.sampleXX.*.network_interface.0.address'
* google_dns_record_set.sampleXX[1]: Resource 'google_compute_instance_from_template.sampleXX' does not have attribute 'network_interface.0.address' for variable 'google_compute_instance_from_template.sampleXX.*.network_interface.0.address'
```
...if I comment out the google_dns_record_set stanza, run terraform to create the VMs, and then add the google_dns_record_set stanza back, things work as expected.

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. set up a 'google' provider with appropriate credentials
2. create config file as above, execute `terraform plan` 

### Important Factoids
None.

### References
None.

**Comment by ndmckinley** [comment]
That's very interesting!  I have a theory about this and I'm working on it right now.

**Comment by elfchief** [comment]
Hmmmm. One thing I didn't mention (but should have) is that if I use `google_compute_instance` instead of `google_compute_instance_from_template`, things work exactly as expected. With regards to the network_interface structure, I'd expect both resource types to have identical structures -- is that not the case?

**Comment by ndmckinley** [comment]
That does feel a little unlikely, doesn't it?  Let me see if I can figure that one out!

**Comment by danawillow** [comment]
@ndmckinley assigning this to you since you've already looked into it a bit, but feel free to unassign if you aren't looking at this anymore.

**Comment by ndmckinley** [comment]
I wasn't able to figure it out back in July.  I still think it's got something to do with the complex `element` stuff.  I'll take one more look.

**Comment by ndmckinley** [comment]
Hm.  I think I have a suggestion for a workaround - it's kind of nasty - the sort of thing I would really expect HCL2 to fix.  I'm still testing, but I _think_ if you add `network_interface {}` to your `google_compute_instance_from_template` resource, it should work.  It works for me with a small patch that I'm trying to refine into a minimal repro case.

**Comment by ndmckinley** [comment]
There we go - once #2018 is merged, you can make this work by adding the same network_interface block from the template (it _has_ to be the same block, otherwise the two will disagree and that causes problems) and downstream `element` functions will be able to access the other computed members of the network_interface block just fine.

**Comment by KIVagant** [comment]
TL;DR:

```
resource "google_compute_instance_template" "default" {
...
  network_interface = {
    network       = "default"
    access_config = {}
  }
...
}

resource "google_compute_instance_from_template" "default" {
  name           = "test"
...
  source_instance_template = "${google_compute_instance_template.default.self_link}"

  network_interface = { // we have to override the template value to get a created IP address in Terraform
    network       = "${var.network_name}"
    access_config = {}
  }
...
}

output "instance_ip" {
  value = "${google_compute_instance_from_template.default.network_interface.0.address}"
}
```
